processing/src/main/java/io/druid/query/groupby/epinephelinae/BufferGrouper.java

```diff
@@ -428,10 +431,17 @@ private void growIfPossible()
     final ByteBuffer entryBuffer = tableBuffer.duplicate();
     final ByteBuffer keyBuffer = tableBuffer.duplicate();
 
+    List<Map<Integer, Integer>> reloationMaps = new ArrayList<>(aggregators.length);
+    for (int i = 0; i < aggregators.length ; i++)
+    {
+      reloationMaps.add(new HashMap<Integer, Integer>(buckets));
```

Comment:
This allocates an on-heap hashmap equal to the size of the off-heap table, which spoils the effort to keep memory usage offheap in groupBy v2. That's especially true for smaller bucket sizes (primitives rather than sketches) where this relocation map might be even larger than the off-heap table.

Please avoid creating `relocationMaps`, you should instead be able to call `relocate` on the BufferAggregator as entries are moved around.